tests/DialogServiceConnectorTests.ts

```diff
@@ -0,0 +1,169 @@
+beforeEach(() => {
+    objsToClose = [];
+    // tslint:disable-next-line:no-console
+    console.info("---------------------------------------Starting test case-----------------------------------");
+});
+
+afterEach(() => {
+    // tslint:disable-next-line:no-console
+    console.info("End Time: " + new Date(Date.now()).toLocaleString());
+    objsToClose.forEach((value: any, index: number, array: any[]) => {
+        if (typeof value.close === "function") {
+            value.close();
+        }
+    });
+});
+
+function BuildDialogServiceConfig(): sdk.DialogServiceConfig {
+    const config: sdk.DialogServiceConfig = sdk.DialogServiceConfig.fromBotSecret(Settings.BotSecret, Settings.SpeechSubscriptionKey, Settings.SpeechRegion);
+
+    if (undefined !== Settings.proxyServer) {
+        config.setProxy(Settings.proxyServer, Settings.proxyPort);
+    }
+
+    expect(config).not.toBeUndefined();
+    return config;
+}
+
+function BuildConnectorFromWaveFile(dialogServiceConfig?: sdk.DialogServiceConfig, audioFileName?: string): sdk.DialogServiceConnector {
+    let connectorConfig: sdk.DialogServiceConfig = dialogServiceConfig;
+    if (connectorConfig === undefined) {
+        connectorConfig = BuildDialogServiceConfig();
+
+        // Since we're not going to return it, mark it for closure.
+        objsToClose.push(connectorConfig);
+    }
+
+    const f: File = WaveFileAudioInput.LoadFile(audioFileName === undefined ? Settings.WaveFile : audioFileName);
+    const audioConfig: sdk.AudioConfig = sdk.AudioConfig.fromWavFileInput(f);
+
+    const language: string = Settings.WaveFileLanguage;
+    if (connectorConfig.speechRecognitionLanguage === undefined) {
+        connectorConfig.speechRecognitionLanguage = language;
+    }
+
+    const connector: sdk.DialogServiceConnector = new sdk.DialogServiceConnector(connectorConfig, audioConfig);
+    expect(connector).not.toBeUndefined();
+
+    return connector;
+}
+
+// DialogServiceConfig tests: begin
+test("Create DialogServiceConfig from bot secret, null params", () => {
+    // tslint:disable-next-line:no-console
+    console.info("Name: Create DialogServiceConfig from bot secret, null params");
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(null, null, null)).toThrowError();
+});
+
+test("Create DialogServiceConfig from bot secret, null subscription and region", () => {
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(Settings.BotSecret, null, null)).toThrowError();
+});
+
+test("Create DialogServiceConfig from bot secret, null botSecret", () => {
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(null, Settings.SpeechSubscriptionKey, null)).toThrowError();
+});
+
+test("Create DialogServiceConfig from bot secret, null Region", () => {
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(Settings.BotSecret, Settings.SpeechSubscriptionKey, null)).toThrowError();
+});
+
+// test("Create DialogServiceConfig from bot secret, valid params", () => {
+
+// })
```

Comment:
?

Reply:
Missed that.. I'm adding more in the next PR, will address there

---
In reply to: [332231472](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/90#discussion_r332231472) [](ancestors = 332231472)